src/mongocxx/options/find.cpp

```diff
@@ -211,6 +214,179 @@ const stdx::optional<class read_preference>& find::read_preference() const {
     return _read_preference;
 }
 
+find& find::convert_all_modifiers() {
+    if (!_modifiers) {
+        return *this;
+    }
+
+    for (auto&& ele : _modifiers->view()) {
+        if (ele.key() == stdx::string_view("$comment")) {
+            if (ele.type() != bsoncxx::type::k_utf8) {
+                throw exception{error_code::k_invalid_parameter,
+                                "string type required for $comment"};
+            }
+            if (!_comment) {
+                _comment = std::string{ele.get_utf8().value.to_string()};
+            }
+        } else if (ele.key() == stdx::string_view("$explain")) {
+            throw exception{error_code::k_invalid_parameter, "$explain modifier unsupported"};
+        } else if (ele.key() == stdx::string_view("$hint")) {
+            switch (ele.type()) {
+                case bsoncxx::type::k_document:
+                    if (!_hint) {
+                        _hint = mongocxx::hint{bsoncxx::document::value{ele.get_document().value}};
+                    }
+                    break;
+                case bsoncxx::type::k_utf8:
+                    if (!_hint) {
+                        _hint = mongocxx::hint{std::string{ele.get_utf8().value.to_string()}};
+                    }
+                    break;
+                default:
+                    throw exception{error_code::k_invalid_parameter,
+                                    "document or string type required for $hint"};
+            }
+        } else if (ele.key() == stdx::string_view("$max")) {
+            if (ele.type() != bsoncxx::type::k_document) {
+                throw exception{error_code::k_invalid_parameter, "document type required for $max"};
+            }
+            if (!_max) {
+                _max = bsoncxx::document::value{ele.get_document().value};
+            }
+        } else if (ele.key() == stdx::string_view("$maxScan")) {
+            std::int32_t max_scan_int32;
+            std::int64_t max_scan_int64;
+            double max_scan_double;
+            switch (ele.type()) {
+                case bsoncxx::type::k_int32:
+                    max_scan_int32 = ele.get_int32().value;
+                    if (max_scan_int32 < 0) {
+                        throw exception{error_code::k_invalid_parameter,
+                                        "non-negative value required for $maxScan"};
+                    }
+                    if (!_max_scan) {
+                        _max_scan = max_scan_int32;
+                    }
+                    break;
+                case bsoncxx::type::k_int64:
+                    max_scan_int64 = ele.get_int64().value;
+                    if (max_scan_int64 < 0 ||
+                        max_scan_int64 > std::numeric_limits<std::int32_t>::max()) {
```

Comment:
I couldn't find any utility functions in the repository for supporting safe numeric conversions.  Should I start something new?